lilypad/server/client.py

```
@@ -271,41 +272,58 @@ def get_prompt_active_version(
-            and generation.prompt.signature == closure.signature
-        ):
-            return generation.prompt
-        elif generation and not generation.prompt:
-            prompts = self._request(
-                "GET",
-                f"v0/projects/{self.project_uuid}/prompts/metadata/signature/public",
-                params={"signature": closure.signature},
-                response_model=list[PromptPublic],
-            )
-            if not prompts:
-                return None
-
-            self._request(
-                "PATCH",
-                f"v0/projects/{self.project_uuid}/generations/{generation.uuid}",
-                json={"prompt_uuid": str(prompts[0].uuid)},
-                response_model=GenerationPublic,
-            )
-
-            return prompts[0]
+        if generation:
+            if generation.prompt and generation.prompt.signature == closure.signature:
+                return generation.prompt
+            elif not generation.prompt:
+                prompts = self._request(
+                    "GET",
+                    f"v0/projects/{self.project_uuid}/prompts/metadata/signature/public",
+                    params={"signature": closure.signature},
+                    response_model=list[PromptPublic],
+                )
+                if prompts:
+                    self._request(
+                        "PATCH",
+                        f"v0/projects/{self.project_uuid}/generations/{generation.uuid}",
+                        json={"prompt_uuid": str(prompts[0].uuid)},
+                        response_model=GenerationPublic,
+                    )
+                    return prompts[0]
         try:
             return self._request(
                 "GET",
                 f"v0/projects/{self.project_uuid}/prompts/hash/{closure.hash}/active",
                 response_model=PromptPublic,
             )
         except NotFoundError:
-            return None
+            prompts = self._request(
```

Do we want to create a empty prompt if we dont find one? Shouldn't we print a link to the UI to create a prompt? @willbakst

---

I think we can print a UI link. Given recent discussion around managed prompt -> managed generation switch, I'm not sure how much this matter anymore.

---

@brenkao 
> Do we want to create a empty prompt if we dont find one?

Do you have any recommended idea?

---

Hmm, I think for now if there is no prompt found when trying to run it, we should just raise an error that tells them to create one with a link to the prompt creation UI

---

OK, I fixed to raise an exception for messages containing URLs when a record is not found.